cmd/agent/api/server_ipc.go

```diff
@@ -0,0 +1,80 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package api
+
+import (
+	"crypto/tls"
+	"encoding/json"
+	"fmt"
+	"net"
+	"net/http"
+	"time"
+
+	gorilla "github.com/gorilla/mux"
+
+	"github.com/DataDog/datadog-agent/pkg/config"
+)
+
+var ipcConfigListener net.Listener
+var allowedConfigPaths = map[string]struct{}{
+	"api_key": {},
+}
+
+func startIPCServer(ipcConfigHostPort string, tlsConfig *tls.Config) (err error) {
+	ipcConfigListener, err = getListener(ipcConfigHostPort)
+	if err != nil {
+		return err
+	}
+
+	configEndpointHandler := func(w http.ResponseWriter, r *http.Request) {
+		vars := gorilla.Vars(r)
+
+		body, err := getConfigMarshalled(vars["path"])
```

Comment:
nit maybe?
```suggestion
		body, err := getConfigValueAsJSON(vars["path"])
```

Reply:
done

Reply:
@pgimalac, the changes do not show in GH. Did you push?